lib/sites/greenman-gaming.js

```diff
@@ -0,0 +1,81 @@
+'use strict';
+
+const siteUtils = require('../site-utils');
+const logger = require('../logger')();
+
+class GreenmanGamingSite {
+  constructor(uri) {
+    if (!GreenmanGamingSite.isSite(uri)) {
+      throw new Error('invalid uri for GreenmanGaming: %s', uri);
+    }
+
+    this._uri = uri;
+  }
+
+  getURIForPageData() {
+    return this._uri;
+  }
+
+  isJSON() {
+    return false;
+  }
+
+  findPriceOnPage($) {
+    let currencyTag = '';
+    let productJson = '';
+    // To find Currency
+    // TODO Do the same with regex!
+    $('script').each(function () {
+      currencyTag = $(this).text().trim();
+      if (currencyTag.indexOf('var utag_data') !== -1) {
+        productJson = currencyTag.substring(currencyTag.indexOf('{'), currencyTag.length - 1);
+        productJson = JSON.parse(productJson);
+      }
+    });
+
+    const currency = productJson.currency_code;
+    const priceString = productJson.product_price_readable;
+    // were we successful?
+    if (!priceString) {
+      logger.error('price not found on GreenmanGaming page, uri: %s', this._uri);
+      return -1;
+    }
+
+    // process the price string
+    const price = siteUtils.processPrice(`${currency} ${priceString}`);
+
+    return price;
+  }
```

Comment:
I'm curious here, from the unit test below it looks like you may have originally looked for the price via the `curPrice` CSS class, was that not a viable solution?  For whatever reason, I'm unable to view the site to see any prices other than dollars, but it looks like it shows different currencies for different regions perhaps?  Again, this is all based on the information in your unit test below.

If it is possible to use the CSS class, that solution looks a bit easier:

```
  findPriceOnPage($) {
    // the various ways we can find the price
    const selectors = [
      '#buy_now .curPrice',
    ];

    // find the price on the page
    const priceString = siteUtils.findContentOnPage($, selectors);

    // were we successful?
    if (!priceString) {
      logger.error('price not found on GreenmanGaming page, uri: %s', this._uri);
      return -1;
    }

    // process the price string
    const price = siteUtils.processPrice(priceString);

    return price;
  }
```

That way we can avoid having to parse that script tag to grab the price and currency.  I'm just curious as to what that produces for different regions where the site shows a different currency than USD.